src/NUnitFramework/framework/Internal/TypeHelper.cs

```diff
@@ -387,5 +388,16 @@ internal static string FullName(this Type type)
         {
             return type.FullName ?? throw new InvalidOperationException("No name for type: " + type);
         }
+
+        private static readonly Type[] EqualsObjectParameterTypes = { typeof(object) };
+
+        internal static bool IsRecord(this Type type)
+        {
+            // Check if Equals method has CompilerGenerated attribute
+            var equalsMethod = type.GetMethod(nameof(type.Equals), BindingFlags.Instance | BindingFlags.Public,
```

Comment:
This tests for `Equals(object)`. It would be better to check for `Equals(T)`. See my `Person` record class. It defined `Equals(Person)` but not `Equals(object)` as the generated version is fine as it defers to `Equals(Person)`.

Reply:
This is only relevant if the class has reference value properties (or an ImmutableArray).
Only for records where the default generated implementation would be wrong.